components/TextFieldsBeta/src/MDCBaseTextField.h

```diff
@@ -0,0 +1,54 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <UIKit/UIKit.h>
+
+/**
+ A UITextField subclass that provides the foundation for Material TextFields.
+ */
+@interface MDCBaseTextField : UITextField
+
+/**
+ This is an RTL-aware wrapper around UITextField's leftView/rightView class.
+ */
+@property(strong, nonatomic, nullable) UIView *leadingView;
+
+/**
+ This is an RTL-aware wrapper around UITextField's leftView/rightView class.
+ */
+@property(strong, nonatomic, nullable) UIView *trailingView;
+
+/**
+ This is an RTL-aware wrapper around UITextField's leftViewMode/rightViewMode class.
+ */
+@property(nonatomic, assign) UITextFieldViewMode leadingViewMode;
+
+/**
+ This is an RTL-aware wrapper around UITextField's leftViewMode/rightViewMode class.
+ */
+@property(nonatomic, assign) UITextFieldViewMode trailingViewMode;
```

Comment:
Should we mark the left/right properties as unavailable?

Reply:
I didn't think about that... Maybe? Doing this right now as a deterrent: https://github.com/material-components/material-components-ios/pull/7410/files#diff-2bb0d7745523c7ad2b75943cfacc8035R70

Reply:
I'm worried about the ramifications of supporting both APIs; it may be clearer for clients if we indicate that this is the preferred mechanism.

Reply:
I don't think we have to worry about supporting the left/right APIs since they're just part of UITextField anyway. Also, their leading/trailing counterparts are just wrappers around them, so marking them as unavailable breaks all the leading/trailing stuff as I currently have it. Building on top of the left/right APIs internally lets us hook into various UITextField layout methods like `-leftViewRectForBounds:` and `-rightViewRectForBounds:`. I think providing the leading/trailing options and adding print statements reminding users of them when they try to use the left/right APIs should be fine.